### PR TITLE
Dark sky caching and migration from joda

### PR DESCRIPTION
**Pull request by asoltysik**
Those two commits packed together because both are related to cache refactor. The cache is now abstracted a bit to the `Cache` class (with functions moved form `CacheUtils` to `Cache` companion object).

### CODE REVIEW

File: src/main/scala/com.snowplowanalytics/weather/Cache.scala [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+   * @param doRequest function which will be called after a cache miss
+   * @return value stored in the cache or the result of the provided function
+   */
+  def getCachedOrRequest(latitude: Float, longitude: Float, dateTime: ZonedDateTime)(
+    doRequest: (Float, Float, ZonedDateTime) => F[Either[WeatherError, W]]): F[Either[WeatherError, W]] = {
+
+    val cacheKey = eventToCacheKey(dateTime, Position(latitude, longitude), geoPrecision)
+    cache.get(cacheKey).flatMap {
+      case Some(Right(cached)) =>
+        Sync[F].delay(Right(cached)) // Cache hit
```
Comment (BenFradet) [inline src/main/scala/com.snowplowanalytics/weather/Cache.scala]: shouldn't this be `pure`

Comment (asoltysik) [reply src/main/scala/com.snowplowanalytics/weather/Cache.scala]: done

File: src/main/scala/com.snowplowanalytics/weather/providers/darksky/DarkSkyCacheClient.scala [hunk 2]
```diff
@@ -0,0 +1,43 @@
+package com.snowplowanalytics.weather
```
Comment (BenFradet) [inline src/main/scala/com.snowplowanalytics/weather/providers/darksky/DarkSkyCacheClient.scala]: missing apache header

Comment (asoltysik) [reply src/main/scala/com.snowplowanalytics/weather/providers/darksky/DarkSkyCacheClient.scala]: Done

**Pull request merged**